### PR TITLE
Limit jfr dump content

### PR DESCRIPTION
**Pull request by laurit**
`FlightRecorder.getFlightRecorder().takeSnapshot()` contains all recorded data which includes events that we have already processed.

### CODE REVIEW

File: profiler/src/test/java/com/splunk/opentelemetry/profiler/JfrRecorderTest.java [hunk 1]
```diff
@@ -51,6 +54,13 @@ void setup() {
     settings.put("foo", "bar");
   }
 
+  @BeforeAll
+  static void setupClass() throws Exception {
+    Path dumpDir = Path.of("build/test-jfr-dump");
+    dumpDir.toFile().mkdirs();
+    outdir = Files.createTempDirectory(dumpDir, "tmp");
```
Comment (mateuszrzeszutek) [inline profiler/src/test/java/com/splunk/opentelemetry/profiler/JfrRecorderTest.java:61]: Would JUnit's `@TempDir` annotation on the field work?

**Comment by breedx-splk** [comment]
So if this were a problem, wouldn't we expect to see overlapping data in the jfr files?  I run the agent with `-Dsplunk.profiler.keep-files=true` and let it build up 5 files....then do this, which this PR suggests should have overlapping/duplicated data.

```
$ for f in $(ls -1tr *.jfr) ; do
> jfr print --events jdk.ThreadDump $f | grep startTime
> done
  startTime = 09:36:26.425
  startTime = 09:36:27.427
  startTime = 09:36:28.432
  startTime = 09:36:29.433
  startTime = 09:36:30.438
  startTime = 09:36:31.440
  startTime = 09:36:32.441
  startTime = 09:36:33.442
  startTime = 09:36:34.444
  startTime = 09:36:35.445
  startTime = 09:36:36.449
  startTime = 09:36:37.454
  startTime = 09:36:38.457
  startTime = 09:36:39.463
  startTime = 09:36:40.469
  startTime = 09:36:41.478
  startTime = 09:36:42.484
  startTime = 09:36:43.487
  startTime = 09:36:44.490
  startTime = 09:36:45.496
  startTime = 09:36:46.502
  startTime = 09:36:47.504
  startTime = 09:36:48.506
  startTime = 09:36:49.507
  startTime = 09:36:50.509
  startTime = 09:36:51.514
  startTime = 09:36:52.519
  startTime = 09:36:53.523
  startTime = 09:36:54.527
  startTime = 09:36:55.532
  startTime = 09:36:56.537
  startTime = 09:36:57.542
  startTime = 09:36:58.546
  startTime = 09:36:59.548
  startTime = 09:37:00.551
  startTime = 09:37:01.557
  startTime = 09:37:02.559
  startTime = 09:37:03.563
  startTime = 09:37:04.570
  startTime = 09:37:05.572
  startTime = 09:37:06.578
  startTime = 09:37:07.583
  startTime = 09:37:08.587
  startTime = 09:37:09.594
  startTime = 09:37:10.597
  startTime = 09:37:11.604
  startTime = 09:37:12.608
  startTime = 09:37:13.614
  startTime = 09:37:14.616
  startTime = 09:37:15.622
  startTime = 09:37:16.624
  startTime = 09:37:17.629
  startTime = 09:37:18.634
  startTime = 09:37:19.639
  startTime = 09:37:20.642
  startTime = 09:37:21.644
  startTime = 09:37:22.648
  startTime = 09:37:23.653
  startTime = 09:37:24.656
  startTime = 09:37:25.659
  startTime = 09:37:26.662
  startTime = 09:37:27.666
  startTime = 09:37:28.672
```

Any thoughts on why we're not seeing duplicates there?

**Comment by laurit** [comment]
Perhaps it only happens when you have another recording in progress, try adding `-XX:+FlightRecorder -XX:StartFlightRecording=filename=myrecording.jfr` to jvm arguments and you should see duplicates.

**Comment by breedx-splk** [comment]
This definitely feels safer and easier to think about anyway...so let's do this. Thanks for the improvement!

**Pull request merged**